File: backend/src/main/java/com/vueadmin/entity/standard/DataStandard.java
package com.vueadmin.entity.standard;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 数据标准视图实体
 */
@Data
@Entity
@Table(name = "std_data_standard")
public class DataStandard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "standard_code", nullable = false, unique = true, length = 100)
    private String standardCode;

    @Column(name = "standard_name", nullable = false, length = 200)
    private String standardName;

    @Column(name = "category_id")
    private Long categoryId;

    @Column(name = "fields_definition", columnDefinition = "TEXT")
    private String fieldsDefinition;

    @Column(name = "status", length = 20)
    private String status = "draft";

    @Column(name = "version")
    private Integer version = 1;

    @Column(name = "publish_time")
    private LocalDateTime publishTime;

    @Column(name = "approval_status", length = 20)
    private String approvalStatus = "pending";

    @Column(name = "approval_by", length = 50)
    private String approvalBy;

    @Column(name = "approval_time")
    private LocalDateTime approvalTime;

    @Column(name = "approval_comment", columnDefinition = "TEXT")
    private String approvalComment;

    @Column(name = "created_by", length = 50)
    private String createdBy;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_by", length = 50)
    private String updatedBy;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/api/standard/dataStandard.ts
import api from '../index'

// 数据标准视图相关接口

export interface FieldDefinition {
  fieldId: number
  fieldCode?: string
  fieldName?: string
  required: boolean
  defaultValue?: string
  sortOrder?: number
}

export interface DataStandard {
  id?: number
  standardCode: string
  standardName: string
  categoryId?: number
  categoryName?: string
  fieldsDefinition?: FieldDefinition[]
  status?: string
  version?: number
  publishTime?: string
  approvalStatus?: string
  approvalBy?: string
  approvalTime?: string
  approvalComment?: string
  createdBy?: string
  createdAt?: string
  updatedBy?: string
  updatedAt?: string
  description?: string
}

export interface DataStandardQuery {
  keyword?: string
  status?: string
  categoryId?: number
  page?: number
  size?: number
  sortBy?: string
  sortDir?: string
}

// 分页查询
export function getDataStandardList(params: DataStandardQuery) {
  return api({
    url: '/data-standard',
    method: 'get',
    params
  })
}

// 根据ID查询
export function getDataStandardById(id: number) {
  return api({
    url: `/data-standard/${id}`,
    method: 'get'
  })
}

// 创建数据标准
export function createDataStandard(data: DataStandard) {
  return api({
    url: '/data-standard',
    method: 'post',
    data
  })
}

// 更新数据标准
export function updateDataStandard(id: number, data: DataStandard) {
  return api({
    url: `/data-standard/${id}`,
    method: 'put',
    data
  })
}

// 删除数据标准
export function deleteDataStandard(id: number) {
  return api({
    url: `/data-standard/${id}`,
    method: 'delete'
  })
}

// 批量删除
export function batchDeleteDataStandard(ids: number[], updatedBy: string) {
  return api({
    url: '/data-standard/batch',
    method: 'delete',
    data: { ids, updatedBy }
  })
}

// 发布数据标准
export function publishDataStandard(id: number) {
  return api({
    url: `/data-standard/${id}/publish`,
    method: 'put'
  })
}

// 归档数据标准
export function archiveDataStandard(id: number) {
  return api({
    url: `/data-standard/${id}/archive`,
    method: 'put'
  })
}

// 审批数据标准
export function approveDataStandard(id: number, approvalBy: string, approved: boolean, comment?: string) {
  return api({
    url: `/data-standard/${id}/approve`,
    method: 'put',
    data: { approvalBy, approved, comment }
  })
}

// 获取统计信息
export function getDataStandardStatistics() {
  return api({
    url: '/data-standard/statistics',
    method: 'get'
  })
}
